android/src/main/java/nl/fhict/happynews/android/controller/SourceController.java

```diff
@@ -0,0 +1,130 @@
+package nl.fhict.happynews.android.controller;
+
+import android.content.Context;
+import com.google.gson.reflect.TypeToken;
+import nl.fhict.happynews.android.SourceManager;
+import nl.fhict.happynews.android.model.Post;
+import nl.fhict.happynews.android.model.Source;
+import nl.fhict.happynews.android.model.SourceSetting;
+import nl.fhict.happynews.android.persistence.PreferenceJsonController;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+/**
+ * Created by Tobi on 08-May-17.
+ */
+public class SourceController {
+
+    private static final String KEY = "SOURCES";
+    private static PreferenceJsonController<SourceSetting> preferences = new PreferenceJsonController<>();
+
+    private static SourceController instance = new SourceController();
+
+    /**
+     * Returns the instance.
+     * @return An instance.
+     */
+    public static SourceController getInstance() {
+        return instance;
+    }
+
+    private SourceController() {
+
+    }
+
+    /**
+     * Initializes the Source Set.
+     * @param context The context.
+     */
+    public void initialize(Context context) {
+        if (preferences.get(context, KEY) == null) {
+            List<SourceSetting> sources = new ArrayList<>();
+            SourceSetting twitterSourceSetting = new SourceSetting("Twitter");
+            SourceSetting quoteSourceSetting = new SourceSetting("quote");
```

Comment:
Aah ok. Let's make the twitter and article source also lowercase to make it more consistent. Only reason they have a capital letter is because of the Adapter<String>, which is changed with this PR.